lib/devise_invalidatable/hooks/invalidatable.rb

```diff
@@ -6,7 +6,7 @@
 # easier to just use our own id.
 Warden::Manager.after_set_user except: :fetch do |user, warden, _|
   UserSession.deactivate(warden.raw_session['auth_id'])
-  warden.raw_session['auth_id'] = user.activate_session
+  warden.raw_session['auth_id'] = user.activate_session(warden)
```

Comment:
I implemented it with even more generic approach, like this:

```ruby
Warden::Manager.after_set_user except: :fetch do |user, warden, opts|
  UserSession.deactivate(warden.raw_session['auth_id'])
  warden.raw_session['auth_id'] = user.activate_session(opts[:session_params])
end
```

I pass `:session_params` as an option to `sign_in`:
```ruby
sign_in @user, event: :authentication, session_params: { external_token: 'some token from omniauth' }
```

It works for me since I use omniauth-only authorization, but may not work for other devise auth strategies.